tests/Unit/ShortcodeTest.php

```diff
@@ -49,4 +49,42 @@ public function testUnclosedCodeTag(): void
             Shortcode::render('[code]test')
         );
     }
+
+    /**
+     * @dataProvider youtubeUrlProvider
+     */
+    public function testAutoEmbedYoutube(string $url, string $expected): void
+    {
+        $this->assertStringContainsString(
+            $expected,
+            Shortcode::render($url)
+        );
+    }
+
+    public function youtubeUrlProvider(): array
+    {
+        // ["given", "expected"]
+        return [
+            // A vanilla URL
+            ['https://www.youtube.com/v/YYOKMUTTDdA', '//www.youtube-nocookie.com/embed/YYOKMUTTDdA'],
+
+            // With explicit video ID "v"
+            ['https://www.youtube.com/watch?v=W3igStIUp8Y', '//www.youtube-nocookie.com/embed/W3igStIUp8Y'],
+            ['https://www.youtube.com/watch?v=SXuIABMlZKc&t=1m30s', '//www.youtube-nocookie.com/embed/SXuIABMlZKc?start=90'],
```

Comment:
This doesn't appear to work if I actually paste the URL into a forum comment. I think it's escaping the ampersand to `&amp;` and tripping up the regex.

Reply:
Excellent find, you're correct on the root cause. I've added a single line to explicitly restore the ampersands in the event that they are escaped - addressed in https://github.com/RetroAchievements/RAWeb/pull/1593/commits/6256f62f0b9f8dac2f5133a5295e1b156cd2a8df.